From fdcba9a06daf69733457a5daf08295925f8ff5c6 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 11 Jul 2015 00:19:08 +0100
Subject: [PATCH] Set XAUTHLOCALHOSTNAME to for local login (hostname change
 issues)

This is a refresh of the following patch from openSUSE:

https://build.opensuse.org/package/view_file/openSUSE:Factory/lightdm/lightdm-xauthlocalhostname-support.patch?expand=1

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/seat-xlocal.c    | 2 +-
 src/x-server-local.c | 5 +++--
 src/x-server.c       | 3 +++
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/seat-xlocal.c b/src/seat-xlocal.c
index 647adc2..1cca1b8 100644
--- a/src/seat-xlocal.c
+++ b/src/seat-xlocal.c
@@ -241,7 +241,7 @@ seat_xlocal_run_script (Seat *seat, DisplayServer *display_server, Process *scri
     path = x_server_local_get_authority_file_path (x_server);
     process_set_env (script, "DISPLAY", x_server_get_address (X_SERVER (x_server)));
     process_set_env (script, "XAUTHORITY", path);
-
+    process_set_env (script, "XAUTHLOCALHOSTNAME", g_get_host_name ());
     SEAT_CLASS (seat_xlocal_parent_class)->run_script (seat, display_server, script);
 }
 
diff --git a/src/x-server-local.c b/src/x-server-local.c
index 1348b17..dfb83be 100644
--- a/src/x-server-local.c
+++ b/src/x-server-local.c
@@ -152,11 +152,12 @@ XServerLocal *
 x_server_local_new (void)
 {
     XServerLocal *self = g_object_new (X_SERVER_LOCAL_TYPE, NULL);
-    gchar hostname[1024], *number, *name;
+    const gchar *hostname;
+    gchar *number, *name;
 
     x_server_set_display_number (X_SERVER (self), x_server_local_get_unused_display_number ());
 
-    gethostname (hostname, 1024);
+    hostname = g_get_host_name ();
     number = g_strdup_printf ("%d", x_server_get_display_number (X_SERVER (self)));
     x_server_set_authority (X_SERVER (self), x_authority_new_cookie (XAUTH_FAMILY_LOCAL, (guint8*) hostname, strlen (hostname), number));
     g_free (number);
diff --git a/src/x-server.c b/src/x-server.c
index ab75e18..14ca795 100644
--- a/src/x-server.c
+++ b/src/x-server.c
@@ -15,6 +15,7 @@
 
 #include "x-server.h"
 #include "configuration.h"
+#include "x-server-local.h"
 
 struct XServerPrivate
 {  
@@ -168,6 +169,8 @@ x_server_connect_session (DisplayServer *display_server, Session *session)
     else
         l_debug (session, "Not setting XDG_VTNR");
 
+    if (IS_X_SERVER_LOCAL (display_server))
+        session_set_env (session, "XAUTHLOCALHOSTNAME", g_get_host_name ());
     session_set_env (session, "DISPLAY", x_server_get_address (X_SERVER (display_server)));
     session_set_xdisplay (session, x_server_get_address (X_SERVER (display_server)));
     session_set_remote_host_name (session, x_server_get_hostname (X_SERVER (display_server)));
-- 
2.4.5

